app/controllers/simplified-account/settings/stripe-details/bank-account/bank-account.controller.test.js

```diff
@@ -0,0 +1,266 @@
+  }
+  next = sinon.spy()
+  bankAccountController[method](req, res, next)
+}
+
+describe('Controller: settings/stripe-details/bank-account', () => {
+  describe('get', () => {
+    before(() => setupTest('get'))
+
+    it('should call the response method', () => {
+      expect(responseStub.called).to.be.true // eslint-disable-line
+    })
+
+    it('should pass req, res and template path to the response method', () => {
+      expect(responseStub.args[0]).to.include(req)
+      expect(responseStub.args[0]).to.include(res)
+      expect(responseStub.args[0]).to.include('simplified-account/settings/stripe-details/bank-account/index')
+    })
+
+    it('should pass context data to the response method', () => {
+      expect(responseStub.args[0][3]).to.have.property('backLink').to.equal(STRIPE_DETAILS_INDEX_PATH)
+      expect(responseStub.args[0][3]).to.have.property('submitLink').to.equal(STRIPE_DETAILS_BANK_ACCOUNT_PATH)
+    })
+  })
+
+  describe('post', () => {
+    const VALID_SORT_CODE = '309430'
+    const VALID_ACCOUNT_NUMBER = '00733445'
+    describe('when submitting valid bank details', () => {
+      before(() => setupTest('post', {}, {}, {
+        body: {
+          sortCodeInput: VALID_SORT_CODE,
+          accountNumberInput: VALID_ACCOUNT_NUMBER
+        }
+      }))
+
+      it('should submit bank details to the stripe details service', () => {
+        expect(updateStripeDetailsBankAccountStub.calledWith(req.service, req.account, VALID_SORT_CODE, VALID_ACCOUNT_NUMBER)).to.be.true // eslint-disable-line
+      })
+
+      it('should redirect to the stripe details index page', () => {
+        expect(res.redirect.calledOnce).to.be.true // eslint-disable-line
+        expect(res.redirect.args[0][0]).to.include(STRIPE_DETAILS_INDEX_PATH)
+      })
+    })
+
+    describe('when validation fails', () => {
+      describe('for empty fields', () => {
+        before(() => {
+          setupTest('post',
+            {},
+            {},
+            { body: { sortCodeInput: '', accountNumberInput: '' } }
+          )
+        })
+
+        it('should render the form with validation errors', () => {
+          expect(responseStub.calledOnce).to.be.true // eslint-disable-line
+          const responseArgs = responseStub.firstCall.args
+          expect(responseArgs[2]).to.equal('simplified-account/settings/stripe-details/bank-account/index')
+          expect(responseArgs[3].errors.summary).to.deep.equal([
+            {
+              href: '#sort-code-input',
+              text: 'Enter a sort code'
+            },
+            {
+              href: '#account-number-input',
+              text: 'Enter an account number'
+            }
+          ])
+          expect(responseArgs[3].errors.formErrors.sortCodeInput).to.equal('Enter a sort code')
+          expect(responseArgs[3].errors.formErrors.accountNumberInput).to.equal('Enter an account number')
+        })
+
+        it('should preserve the user input', () => {
```

Comment:
Is this test name correct? It's under the category "for empty fields" so there's no user input.